assets/src/block-editor/helpers/index.js

```diff
@@ -354,23 +354,68 @@ export const setImageBlockLayoutAttributes = ( props, layout ) => {
 /**
  * Show notice if video player have autoplay enabled and mute is disabled.
  *
- * @param {Object} props Props.
+ * @param {Object}  props                     Props.
+ * @param {boolean} props.isSelected          Flag indicating if the block selected.
+ * @param {Object}  props.attributes          Block attributes.
+ * @param {boolean} props.attributes.autoplay Flag indicating if the video has autoplay turned on.
+ * @param {boolean} props.attributes.muted    Flag indicating if the video is muted.
  * @return {ReactElement} Inspector Controls.
  */
-export const setUpVideoInspectorControls = ( props ) => {
-	const { isSelected, attributes: { autoplay, muted } } = props;
+export const VideoInspectorControls = ( props ) => {
+	const { attributes: { autoplay, muted } } = props;
+
+	const message = useRef( null );
+	const blockTypeDescription = useSelect( ( _select ) => _select( 'core/blocks' ).getBlockType( 'core/video' )?.description, [] );
+
+	/**
+	 * Since we are mutating the DOM, it's not the regular `useEffect` hook.
+	 */
+	useLayoutEffect( () => {
+		const showMessage = true === autoplay && true !== muted;
+
+		if ( showMessage && ! message.current ) {
+			/**
+			 * When switching over from another block selection, the message is
+			 * tried to be added before the sidebar content changes. This delay
+			 * prevents this issue.
+			 */
+			setTimeout( () => {
+				/**
+				 * Since there is no actual ID or block type designator exposed
+				 * in the block editor's sidebar DOM, we need to use to a hack.
+				 * In order to determine the video settings DOM container we are
+				 * first searching for a block type description. Once we have it
+				 * we traverse through the DOM tree to finally find out the
+				 * "Muted" toggle container.
+				 */
+				const blockInspector = [ ...document.querySelectorAll( '.edit-post-sidebar .block-editor-block-card__description' ) ]
+					?.find( ( description ) => description.textContent === blockTypeDescription )?.closest( '.block-editor-block-inspector' );
+
+				if ( ! blockInspector ) {
+					return;
+				}
 
-	if ( ! isSelected ) {
-		return null;
-	}
+				const mutedToggleContainerIndex = 3;
+				const mutedToggleContainer = blockInspector.querySelector( '.editor-video-poster-control' )?.parentElement.children[ mutedToggleContainerIndex ];
+
+				if ( ! mutedToggleContainer ) {
+					return;
+				}
+
+				message.current = document.createElement( 'p' );
+				message.current.classList.add( 'amp-video-autoplay-notice' );
+				message.current.textContent = __( 'Autoplay will mute the video player by default in AMP mode.', 'amp' );
```

Comment:
```suggestion
				message.current.textContent = __( 'Autoplay will cause the video to be muted in many browsers to prevent a poor user experience. It will be muted in AMP for this reason as well.', 'amp' );
```

See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/video#:~:text=In%20some%20browsers%20(e.g.%20Chrome%2070.0)%20autoplay%20doesn%27t%20work%20if%20no%20muted%20attribute%20is%20present.

Reply:
Addressed in 9c80afd